src/arc.ts

```diff
@@ -214,29 +214,35 @@ export class Arc {
   }
 
   public getObservable(query: any, apolloQueryOptions: IApolloQueryOptions = {}) {
-    Logger.debug(query.loc.source.body)
 
-    const subscriptionQuery = gql`
-      subscription ${query}
-    `
-    const zenObservable: ZenObservable<object[]> = this.apolloClient.subscribe<object[]>({ query: subscriptionQuery })
-    const subscriptionObservable = Observable.create((observer: Observer<object[]>) => {
-      const subscription = zenObservable.subscribe(observer)
-      return () => subscription.unsubscribe()
-    })
+    return Observable.create((observer: Observer<ApolloQueryResult<any>>) => {
+      Logger.debug(query.loc.source.body)
 
-    const queryPromise: Promise<ApolloQueryResult<{[key: string]: object[]}>> = this.apolloClient.query(
-      { query, ...apolloQueryOptions })
+      // we query directly to get the current state
+      const queryPromise: Promise<ApolloQueryResult<{[key: string]: object[]}>> = this.apolloClient.query(
+        { query, ...apolloQueryOptions })
 
-    const queryObservable = from(queryPromise).pipe(
-      concat(subscriptionObservable)
-    ).pipe(
-      catchError((err: Error) => {
-        throw Error(`${err.name}: ${err.message}\n${query.loc.source.body}`)
-      })
-    )
+      const subscriptionQuery = gql`
+          subscription ${query}
+        `
+      const zenObservable: ZenObservable<object[]> = this.apolloClient.subscribe<object[]>({ query: subscriptionQuery })
+      const subscriptionObservable = Observable.create((obs: Observer<any>) => {
+          const subscription = zenObservable.subscribe(obs)
+          return () => subscription.unsubscribe()
+        })
 
-    return queryObservable as Observable<any>
+      const sub = from(queryPromise)
```

Comment:
this code is confusing, would be nice to have more comments throughout

Reply:
ok